Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift

```diff
@@ -131,18 +131,23 @@ open class ChatMessageActionsVC: _ViewController, ThemeProvider {
 
             if canDeleteAnyMessage {
                 actions.append(deleteActionItem())
-            } else if canDeleteOwnMessage && message.isSentByCurrentUser {
+            } else if canDeleteOwnMessage && isSentByCurrentUser {
                 actions.append(deleteActionItem())
             }
 
             if !isSentByCurrentUser {
                 actions.append(flagActionItem())
             }
 
-            if channelConfig.mutesEnabled && !message.isSentByCurrentUser {
+            if channelConfig.mutesEnabled && !isSentByCurrentUser {
                 let isMuted = currentUser.mutedUsers.map(\.id).contains(message.author.id)
                 actions.append(isMuted ? unmuteActionItem() : muteActionItem())
             }
+            
+            if !isSentByCurrentUser {
+                let isBlocked = currentUser.blockedUsers.map(\.userId).contains(message.author.id)
+                actions.append(isBlocked ? unblockActionItem() : blockActionItem())
+            }
```

Comment:
We should not introduce this by default because it will break the customer's UI, and some customers might not even want this feature. So we need to introduce a Components flag, like `isBlockingUsersEnabled`. Or we don't have put this in our SDK, and only in the DemoApp 🤔 Depends on the requirements

Reply:
Added `isBlockingUsersEnabled` 👍